### PR TITLE
Add instrumentation tests for browser sign in

### PR DESCRIPTION
**Pull request by rajdeepnanua-okta**

### CODE REVIEW

File: browser-sign-in/app/build.gradle [hunk 1]
```diff
@@ -51,6 +60,8 @@ dependencies {
     androidTestImplementation(libs.jackson.dataformat.yml)
     androidTestImplementation(libs.bundles.android.instrumentation.testing)
     androidTestImplementation(libs.bundles.okta.testing)
+    androidTestUtil(libs.androidx.test.orchestrator)
+    debugImplementation(libs.androidx.test.fragment)
```
Comment (JayNewstrom) [inline browser-sign-in/app/build.gradle:64]:
```suggestion
    androidTestImplementation(libs.androidx.test.fragment)
```

File: browser-sign-in/app/src/androidTest/java/com/okta/android/samples/browser_sign_in/user_dashboard/DashboardPage.kt [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    fun assertUserInfoContainsNameAndEmail(name: String, email: String): DashboardPage {
+        waitForView(withId(R.id.user_info_text_view), 10_000L)
+        with(onView(withId(R.id.user_info_text_view))) {
+            check(matches(withText(containsString(name))))
+            check(matches(withText(containsString(email))))
+        }
+        return this
+    }
+
+    fun assertUserInfoContainsExpectedClaims(): DashboardPage {
```
Comment (JayNewstrom) [inline browser-sign-in/app/src/androidTest/java/com/okta/android/samples/browser_sign_in/user_dashboard/DashboardPage.kt:63]: Can we assert that it also contains `sub: ${userRule.email}`?

**Review by JayNewstrom — APPROVED** [review]
Overall this looks great! Nice work!

2 small comments, but not blockers.

File: browser-sign-in/app/build.gradle (hunk 1 above)
Comment (rajdeepnanua-okta) [reply browser-sign-in/app/build.gradle:64]: This is actually correct. It's super unintuitive, and not sure why exactly it's supposed to be this way, but https://developer.android.com/guide/fragments/test says it should be imported like this. And setting this to debugImplementation fixed an issue with my tests locally where the test wasn't redirecting to my activity after signing into the browser.

File: browser-sign-in/app/src/androidTest/java/com/okta/android/samples/browser_sign_in/user_dashboard/DashboardPage.kt (hunk 2 above)
Comment (rajdeepnanua-okta) [reply browser-sign-in/app/src/androidTest/java/com/okta/android/samples/browser_sign_in/user_dashboard/DashboardPage.kt:63]: Yep, will do

**Review by JayNewstrom — APPROVED** [review]

**Pull request merged**